src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -790,6 +790,123 @@ public StringBuilder Append(string value, int startIndex, int count)
             }
         }
 
+        public StringBuilder Append(StringBuilder value)
```

Comment:
it is not important but I find it slightly easier to read if the equals true case is first. @AndyAyersMS does it matter for code gen if we put the common branch second? or does it pick the ordering itself?
```c#
               if (value == this)
                    Append(value.ToString());
               else
                    AppendCore(value, 0, value.Length);
```

Reply:
It should not matter. Pick whichever version is most readable.